src/main/java/gregtech/api/recipes/logic/ParallelLogic.java

```diff
@@ -440,7 +440,12 @@ public static RecipeBuilder<?> doParallelRecipes(@Nonnull Recipe currentRecipe,
         if (multiplierByInputs == 0) {
             return null;
         }
-        RecipeBuilder<?> recipeBuilder = recipeMap.recipeBuilder();
+        // Make a copy of the recipe builder and zero the EUt, since we append
+        // the total multiplied EUt, and not doing so may add an extra multiple
+        // for the EUt (for example, x2 recipes but x3 EUt) if the original
+        // recipe builder already has a cost applied. Don't also zero the
+        // duration as it doesn't get multiplied.
+        RecipeBuilder<?> recipeBuilder = recipeMap.recipeBuilder().EUt(0);
```

Comment:
This works in current situation. But I don't think this is correct way how to handle this problem. As you are modifying Builder of provided `recipeMap` in way someone later in line will not expect. Which may cause serious problems which will be hard to track.

Reply:
Does `RecipeMap#recipeBuilder()` not make a copy? Other bits I've read, and the code, suggests it does.

Reply:
I missed that info when I checked it previously. In that case there won't be any problem with modifying current `recipeMap`. And this should be OK.